runtime/lua/vim/treesitter/_query_linter.lua

```diff
@@ -0,0 +1,198 @@
+local namespace = vim.api.nvim_create_namespace('vim.treesitter.query_linter')
+-- those node names exist for every language
+local BUILT_IN_NODE_NAMES = { '_', 'ERROR' }
+
+local M = {}
+
+---@private
+--- Caches parse results for queries for each language.
+--- Entries of parse_cache[lang][query_text] will either be true for successful parse or contain the
+--- error message of the parse
+local parse_cache = {}
+
+---@private
+--- Adds a diagnostic for node in the query buffer
+local function add_lint_for_node(diagnostics, node, buf, lint, lang)
+  local node_text = vim.treesitter.get_node_text(node, buf):gsub('\n', ' ')
+  local message = lint .. ': ' .. node_text
+  local error_range = { node:range() }
+  table.insert(diagnostics, {
+    lnum = error_range[1],
+    end_lnum = error_range[3],
+    col = error_range[2],
+    end_col = error_range[4],
+    severity = vim.diagnostic.ERROR,
+    message = message,
+    source = lang,
+  })
+end
+
+---@private
+--- like vim.tbl_contains but works with function predicate
+--- TODO: Should be handled by standard vim.* functions!
```

Comment:
IMO we should either fix this TODO now or go ahead and create an open issue for it, or it will just languish here forever. Might be a `good-first-issue` candidate.

Reply:
Needs some discussion/decision first, though (new function vs. extending old function).

Reply:
https://github.com/neovim/neovim/issues/23024

Reply:
https://github.com/neovim/neovim/pull/23040

Reply:
That PR is merged, so you can use
```lua
vim.tbl_contains(parser_info.symbols, function(t)
                     return node_type == t[1] and is_named == t[2]
                   end, { predicate = true })
```
now (note the argument order).